spinalcordtoolbox/scripts/sct_run_batch.py

```diff
@@ -40,6 +41,44 @@
 csi_filter.register_codec()
 
 
+class ParseYMLAction(argparse.Action):
+    """Reads in a YML file and determines if it contains subjects and/or files."""
+    def __call__(self, parser, namespace, values, option_string=None):
+        with open(values, 'r') as yaml_file:
+            yaml_contents = yaml.safe_load(yaml_file)
```

Comment:
Note that `parser.parse_args()` only handles exceptions of type `argparse.ArgumentError` (for which it displays the script usage string, and the error message itself). All other exception types (file not found, invalid YAML, ValueErrors, etc.) instead result in a big traceback. Is this what we want?

If we do want to use `argparse.ArgumentError`, note that it should be instantiated as `raise argparse.ArgumentError(self, "some message")` from inside the `__call__(self, ...)` of an `argparse.Action` subclass. Unlike most exception types, which only need `"some message"` and no `self`.

This comment also applies to the 3 instances of `raise ValueError` below.